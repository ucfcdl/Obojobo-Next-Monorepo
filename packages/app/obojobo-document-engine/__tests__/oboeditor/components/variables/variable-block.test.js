import React from 'react'
import { shallow } from 'enzyme'

import VariableBlock from '../../../../src/scripts/oboeditor/components/variables/variable-block'

describe('VariableBlock', () => {
	test('VariableBlock component is selected"', () => {
		const variable = {
			name: 'mock_var',
			type: 'static-value',
			value: '3'
		}

		const component = shallow(
			<VariableBlock
				variable={variable}
				index={0}
				currSelect={0}
				creatingVariable={false}
				onClickVarible={jest.fn()}
				isSelected={true}
			/>
		)

		expect(component.find('.single-variable').hasClass('variable-is-selected')).toEqual(true)

		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('VariableBlock with type "static-value"', () => {
		const variable = {
			name: 'mock_var',
			type: 'static-value',
			value: '3'
		}

		const component = shallow(
			<VariableBlock
				variable={variable}
				index={0}
				currSelect={0}
				creatingVariable={false}
				onClickVarible={jest.fn()}
			/>
		)
		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('VariableBlock with type "static-value" and no default value', () => {
		const variable = {
			name: 'mock_var',
			type: 'static-value'
		}

		const component = shallow(
			<VariableBlock
				variable={variable}
				index={0}
				currSelect={0}
				creatingVariable={false}
				onClickVarible={jest.fn()}
			/>
		)
		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('VariableBlock with type "random-number"', () => {
		const variable = {
			name: 'b',
			type: 'random-number',
			start: '9',
			valueMax: '10',
			valueMin: '3',
			decimalPlacesMax: '4',
			decimalPlacesMin: '4'
		}

		const component = shallow(
			<VariableBlock
				variable={variable}
				index={0}
				currSelect={0}
				creatingVariable={false}
				onClickVarible={jest.fn()}
			/>
		)
		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('VariableBlock with type "random-number" with no default value', () => {
		const variable = {
			name: 'b',
			type: 'random-number'
		}

		const component = shallow(
			<VariableBlock
				variable={variable}
				index={0}
				currSelect={0}
				creatingVariable={false}
				onClickVarible={jest.fn()}
			/>
		)
		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('VariableBlock with type "static-list"', () => {
		const variable = {
			name: 'c',
			type: 'static-list',
			value: '4, 5, 6'
		}

		const component = shallow(
			<VariableBlock
				variable={variable}
				index={0}
				currSelect={0}
				creatingVariable={false}
				onClickVarible={jest.fn()}
			/>
		)
		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('VariableBlock with type "static-list" with no default value', () => {
		const variable = {
			name: 'c',
			type: 'static-list'
		}

		const component = shallow(
			<VariableBlock
				variable={variable}
				index={0}
				currSelect={0}
				creatingVariable={false}
				onClickVarible={jest.fn()}
			/>
		)
		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('VariableBlock with type "random-list"', () => {
		const variable = {
			name: 'd',
			type: 'random-list',
			unique: 'false',
			sizeMax: '5',
			sizeMin: '3',
			valueMax: '10',
			valueMin: '3',
			decimalPlacesMax: '1',
			decimalPlacesMin: '1'
		}

		const component = shallow(
			<VariableBlock
				variable={variable}
				index={0}
				currSelect={0}
				creatingVariable={false}
				onClickVarible={jest.fn()}
			/>
		)
		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('VariableBlock with type "random-list" with no default value', () => {
		const variable = {
			name: 'd',
			type: 'random-list'
		}

		const component = shallow(
			<VariableBlock
				variable={variable}
				index={0}
				currSelect={0}
				creatingVariable={false}
				onClickVarible={jest.fn()}
			/>
		)
		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('VariableBlock with type "random-sequence"', () => {
		const variable = {
			name: 'e',
			step: '1.1',
			type: 'random-sequence',
			sizeMax: '10',
			sizeMin: '3',
			valueMax: '100',
			valueMin: '1',
			seriesType: 'geometric'
		}

		const component = shallow(
			<VariableBlock
				variable={variable}
				index={0}
				currSelect={0}
				creatingVariable={false}
				onClickVarible={jest.fn()}
			/>
		)
		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('VariableBlock with type "random-sequence" with no default value', () => {
		const variable = {
			name: 'e',
			type: 'random-sequence'
		}

		const component = shallow(
			<VariableBlock
				variable={variable}
				index={0}
				currSelect={0}
				creatingVariable={false}
				onClickVarible={jest.fn()}
			/>
		)
		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('VariableBlock with type "pick-one"', () => {
		const variable = {
			name: 'f',
			type: 'pick-one',
			value: '3, 4, 5, 3, 5'
		}

		const component = shallow(
			<VariableBlock
				variable={variable}
				index={0}
				currSelect={0}
				creatingVariable={false}
				onClickVarible={jest.fn()}
			/>
		)
		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('VariableBlock with type "pick-one" with no default value', () => {
		const variable = {
			name: 'f',
			type: 'pick-one'
		}

		const component = shallow(
			<VariableBlock
				variable={variable}
				index={0}
				currSelect={0}
				creatingVariable={false}
				onClickVarible={jest.fn()}
			/>
		)
		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('VariableBlock with type "pick-list"', () => {
		const variable = {
			name: 'g',
			type: 'pick-list',
			value: '33, 3, 4, 55, 23, 444',
			ordered: 'false',
			valueMax: '40',
			valueMin: '5'
		}

		const component = shallow(
			<VariableBlock
				variable={variable}
				index={0}
				currSelect={0}
				creatingVariable={false}
				onClickVarible={jest.fn()}
			/>
		)
		const tree = component.html()
		expect(tree).toMatchSnapshot()
	})

	test('VariableBlock with type "pick-list" with no default value', () => {
		let variable = {
			name: 'g',
			type: 'pick-list'
		}

		let component = shallow(
			<VariableBlock
				variable={variable}
				index={0}
				currSelect={0}
				creatingVariable={false}
				onClickVarible={jest.fn()}
			/>
		)
		const tree = component.html()
		expect(tree).toMatchSnapshot()

		variable = {
			name: 'g',
			type: 'pick-list',
			value: '3'
		}

		component = shallow(
			<VariableBlock
				variable={variable}
				index={0}
				currSelect={0}
				creatingVariable={false}
				onClickVarible={jest.fn()}
			/>
		)
	})
})
