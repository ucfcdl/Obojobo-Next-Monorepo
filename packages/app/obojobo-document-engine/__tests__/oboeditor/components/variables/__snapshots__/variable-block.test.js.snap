// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VariableBlock VariableBlock component is selected" 1`] = `"<div class=\\"single-variable variable-is-selected\\" tabindex=\\"0\\"><h4>$mock_var</h4><small><span>Static value </span><b>3</b></small></div>"`;

exports[`VariableBlock VariableBlock with type "pick-list" 1`] = `"<div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$g</h4><small><b>5-40 items from [33,  3,  4,  55,  23,  444]</b></small></div>"`;

exports[`VariableBlock VariableBlock with type "pick-list" with no default value 1`] = `"<div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$g</h4><small>Pick list </small></div>"`;

exports[`VariableBlock VariableBlock with type "pick-list" with no default value 2`] = `"<div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$g</h4><small></small></div>"`;

exports[`VariableBlock VariableBlock with type "pick-one" 1`] = `"<div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$f</h4><small><span>Pick from </span><b>[3,  4,  5,  3,  5]</b></small></div>"`;

exports[`VariableBlock VariableBlock with type "pick-one" with no default value 1`] = `"<div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$f</h4><small><span>Pick from </span><b>[]</b></small></div>"`;

exports[`VariableBlock VariableBlock with type "random-list" 1`] = `"<div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$d</h4><small><span>Random list </span><b>3-5 items of 3-10</b></small></div>"`;

exports[`VariableBlock VariableBlock with type "random-list" with no default value 1`] = `"<div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$d</h4><small><span>Random list </span></small></div>"`;

exports[`VariableBlock VariableBlock with type "random-number" 1`] = `"<div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$b</h4><small><span>Random number </span><b>(3-10)</b></small></div>"`;

exports[`VariableBlock VariableBlock with type "random-number" with no default value 1`] = `"<div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$b</h4><small><span>Random number </span></small></div>"`;

exports[`VariableBlock VariableBlock with type "random-sequence" 1`] = `"<div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$e</h4><small><span>Random seq </span><b>3-10 items starting from 1</b></small></div>"`;

exports[`VariableBlock VariableBlock with type "random-sequence" with no default value 1`] = `"<div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$e</h4><small><span>Random seq </span></small></div>"`;

exports[`VariableBlock VariableBlock with type "static-list" 1`] = `"<div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$c</h4><small><b>[4,  5,  6]</b></small></div>"`;

exports[`VariableBlock VariableBlock with type "static-list" with no default value 1`] = `"<div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$c</h4><small><b>[]</b></small></div>"`;

exports[`VariableBlock VariableBlock with type "static-value" 1`] = `"<div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$mock_var</h4><small><span>Static value </span><b>3</b></small></div>"`;

exports[`VariableBlock VariableBlock with type "static-value" and no default value 1`] = `"<div class=\\"single-variable\\" tabindex=\\"0\\"><h4>$mock_var</h4><small><span>Static value </span><b></b></small></div>"`;
